### PR TITLE
Create budget app

### PR DESCRIPTION
**Pull request by shasherazi**
In this PR, I 
- Added splash screen
- Added sign-up and sign-in pages
- Added Home page (categories page)
- Added Transactions page
- Added "Add a new category" page
- Added "Add a new transaction" page
- Added model and system specs

Here is the [live demo link](https://budget-app-te60.onrender.com/), and here is the [loom video link](https://www.loom.com/share/86136be180d941b5b4f5b6a4079c4a80?sid=36ce33e6-85b9-4ddf-82d6-06880415f71e)

You can use the following command to generate the sample user data for testing.
```
rake db:populate
```
The credentials for the test user are:
Email: `testuser@test.com`
Password: `password`

### CODE REVIEW

File: spec/models/user_spec.rb
Comment (franclobo) [inline spec/models/user_spec.rb:1]: - [x] There are some failures with the specs, please fix that to pass all the tests. 👇🏽 

```ruby
franclobo@DESKTOP-E09US2D:~/microverse/budget-app$ rspec spec .
..............
Creating test data...
Logging in...
F
Creating test data...
Logging in...
F
Creating test data...
Logging in...
F
Cleaning up...

Creating test data...
Logging in...
F
Creating test data...
Logging in...
F
Creating test data...
Logging in...
F
Cleaning up...
FFF

Failures:

  1) Entities displays all entities and their names, and dates
     Failure/Error: <%= stylesheet_link_tag "tailwind", "inter-font", "data-turbo-track": "reload" %>
     
     ActionView::Template::Error:
       The asset "tailwind.css" is not present in the asset pipeline.
     # ./app/views/layouts/application.html.erb:8:in `_app_views_layouts_application_html_erb___1174942974728850553_31900'
     # ./spec/system/entities_spec.rb:32:in `block (2 levels) in <main>'
     # ------------------
     # --- Caused by: ---
     # Sprockets::Rails::Helper::AssetNotFound:
     #   The asset "tailwind.css" is not present in the asset pipeline.
     #   ./app/views/layouts/application.html.erb:8:in `_app_views_layouts_application_html_erb___1174942974728850553_31900'

  2) Entities can create a new entity
     Failure/Error: <%= stylesheet_link_tag "tailwind", "inter-font", "data-turbo-track": "reload" %>
     
     ActionView::Template::Error:
       The asset "tailwind.css" is not present in the asset pipeline.
     # ./app/views/layouts/application.html.erb:8:in `_app_views_layouts_application_html_erb___1174942974728850553_31900'
     # ./spec/system/entities_spec.rb:32:in `block (2 levels) in <main>'
     # ------------------
     # --- Caused by: ---
     # Sprockets::Rails::Helper::AssetNotFound:
     #   The asset "tailwind.css" is not present in the asset pipeline.
     #   ./app/views/layouts/application.html.erb:8:in `_app_views_layouts_application_html_erb___1174942974728850553_31900'

  3) Entities can create a new entity with different groups
     Failure/Error: <%= stylesheet_link_tag "tailwind", "inter-font", "data-turbo-track": "reload" %>
     
     ActionView::Template::Error:
       The asset "tailwind.css" is not present in the asset pipeline.
     # ./app/views/layouts/application.html.erb:8:in `_app_views_layouts_application_html_erb___1174942974728850553_31900'
     # ./spec/system/entities_spec.rb:32:in `block (2 levels) in <main>'
     # ------------------
     # --- Caused by: ---
     # Sprockets::Rails::Helper::AssetNotFound:
     #   The asset "tailwind.css" is not present in the asset pipeline.
     #   ./app/views/layouts/application.html.erb:8:in `_app_views_layouts_application_html_erb___1174942974728850553_31900'

  4) Groups displays all groups, their names, and icons
     Failure/Error: <%= stylesheet_link_tag "tailwind", "inter-font", "data-turbo-track": "reload" %>
     
     ActionView::Template::Error:
       The asset "tailwind.css" is not present in the asset pipeline.
     # ./app/views/layouts/application.html.erb:8:in `_app_views_layouts_application_html_erb___1174942974728850553_31900'
     # ./spec/system/groups_spec.rb:32:in `block (2 levels) in <main>'
     # ------------------
     # --- Caused by: ---
     # Sprockets::Rails::Helper::AssetNotFound:
     #   The asset "tailwind.css" is not present in the asset pipeline.
     #   ./app/views/layouts/application.html.erb:8:in `_app_views_layouts_application_html_erb___1174942974728850553_31900'

  5) Groups displays entities of a group when clicked
     Failure/Error: <%= stylesheet_link_tag "tailwind", "inter-font", "data-turbo-track": "reload" %>
     
     ActionView::Template::Error:
       The asset "tailwind.css" is not present in the asset pipeline.
     # ./app/views/layouts/application.html.erb:8:in `_app_views_layouts_application_html_erb___1174942974728850553_31900'
     # ./spec/system/groups_spec.rb:32:in `block (2 levels) in <main>'
     # ------------------
     # --- Caused by: ---
     # Sprockets::Rails::Helper::AssetNotFound:
     #   The asset "tailwind.css" is not present in the asset pipeline.
     #   ./app/views/layouts/application.html.erb:8:in `_app_views_layouts_application_html_erb___1174942974728850553_31900'

  6) Groups can create a new group
     Failure/Error: <%= stylesheet_link_tag "tailwind", "inter-font", "data-turbo-track": "reload" %>
     
     ActionView::Template::Error:
       The asset "tailwind.css" is not present in the asset pipeline.
     # ./app/views/layouts/application.html.erb:8:in `_app_views_layouts_application_html_erb___1174942974728850553_31900'
     # ./spec/system/groups_spec.rb:32:in `block (2 levels) in <main>'
     # ------------------
     # --- Caused by: ---
     # Sprockets::Rails::Helper::AssetNotFound:
     #   The asset "tailwind.css" is not present in the asset pipeline.
     #   ./app/views/layouts/application.html.erb:8:in `_app_views_layouts_application_html_erb___1174942974728850553_31900'

  7) Welcome displays welcome message
     Failure/Error: <%= stylesheet_link_tag "tailwind", "inter-font", "data-turbo-track": "reload" %>
     
     ActionView::Template::Error:
       The asset "tailwind.css" is not present in the asset pipeline.
     # ./app/views/layouts/application.html.erb:8:in `_app_views_layouts_application_html_erb___1174942974728850553_31900'
     # ./spec/system/welcome_spec.rb:8:in `block (2 levels) in <main>'
     # ------------------
     # --- Caused by: ---
     # Sprockets::Rails::Helper::AssetNotFound:
     #   The asset "tailwind.css" is not present in the asset pipeline.
     #   ./app/views/layouts/application.html.erb:8:in `_app_views_layouts_application_html_erb___1174942974728850553_31900'

  8) Welcome displays sign up link
     Failure/Error: <%= stylesheet_link_tag "tailwind", "inter-font", "data-turbo-track": "reload" %>
     
     ActionView::Template::Error:
       The asset "tailwind.css" is not present in the asset pipeline.
     # ./app/views/layouts/application.html.erb:8:in `_app_views_layouts_application_html_erb___1174942974728850553_31900'
     # ./spec/system/welcome_spec.rb:13:in `block (2 levels) in <main>'
     # ------------------
     # --- Caused by: ---
     # Sprockets::Rails::Helper::AssetNotFound:
     #   The asset "tailwind.css" is not present in the asset pipeline.
     #   ./app/views/layouts/application.html.erb:8:in `_app_views_layouts_application_html_erb___1174942974728850553_31900'

  9) Welcome displays log in link
     Failure/Error: <%= stylesheet_link_tag "tailwind", "inter-font", "data-turbo-track": "reload" %>
     
     ActionView::Template::Error:
       The asset "tailwind.css" is not present in the asset pipeline.
     # ./app/views/layouts/application.html.erb:8:in `_app_views_layouts_application_html_erb___1174942974728850553_31900'
     # ./spec/system/welcome_spec.rb:18:in `block (2 levels) in <main>'
     # ------------------
     # --- Caused by: ---
     # Sprockets::Rails::Helper::AssetNotFound:
     #   The asset "tailwind.css" is not present in the asset pipeline.
     #   ./app/views/layouts/application.html.erb:8:in `_app_views_layouts_application_html_erb___1174942974728850553_31900'

Finished in 4.12 seconds (files took 5.17 seconds to load)
23 examples, 9 failures

Failed examples:

rspec ./spec/system/entities_spec.rb:45 # Entities displays all entities and their names, and dates
rspec ./spec/system/entities_spec.rb:53 # Entities can create a new entity
rspec ./spec/system/entities_spec.rb:63 # Entities can create a new entity with different groups
rspec ./spec/system/groups_spec.rb:45 # Groups displays all groups, their names, and icons
rspec ./spec/system/groups_spec.rb:53 # Groups displays entities of a group when clicked
rspec ./spec/system/groups_spec.rb:65 # Groups can create a new group
rspec ./spec/system/welcome_spec.rb:7 # Welcome displays welcome message
rspec ./spec/system/welcome_spec.rb:12 # Welcome displays sign up link
rspec ./spec/system/welcome_spec.rb:17 # Welcome displays log in link
```

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @shasherazi, 

# Suggested changes

istockphoto-1079018918-612x612

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!

## Highlights

- No linter errors ✔️ 
- Descriptive PR ✔️ 
- Correct Git flow ✔️ 
- Use, device for authentication ✔️ 
- Have unit tests ✔️ 

## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: spec/models/user_spec.rb
Comment (shasherazi) [reply spec/models/user_spec.rb:1]: Hi @franclobo. Seeing the logs, you seem to be missing the Windows version of the tailwind-related gems. I think you are using Windows, and I generated the gemfile on my linux system. See [this issue](https://github.com/rails/tailwindcss-rails/issues/153) to see how to add Windows versions of gems. You can also see more info in [this StackOverflow answer](https://stackoverflow.com/a/8699106/14725559).

It is just my guess of what's happening. Bundler provides this flag for adding platforms, as shown in the above link. To assure you, the tests run fine on my system.

image

File: README.md [hunk 1]
```diff
@@ -1,24 +1,136 @@
...
+
+I would like to thank the Microverse community for their support and guidance.
+
+Special thanks for [Gregoire Vella](https://www.behance.net/gregoirevella) by providing the orignal design idea.
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## 📝 License <a name="license"></a>
+
+This project is [MIT](./LICENSE) licensed.
```
Comment (cynthiainga) [inline README.md:134]: - [Optional] Please kindly fix your MIT license link, the link is broken 😁 😁

**Review by cynthiainga — APPROVED** [review]
Hi @shasherazi, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

- [ ] All tests are passing ✔️ 
- [ ] Good design ✔️ 
- [ ] Good commit history ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**